streams/src/main/java/org/apache/kafka/streams/state/internals/InMemoryKeyValueStore.java

```diff
@@ -133,6 +134,16 @@ public void putAll(final List<KeyValue<Bytes, byte[]>> entries) {
             new InMemoryKeyValueIterator(map.entrySet().iterator()));
     }
 
+    @Override
+    public KeyValueIterator<Bytes, byte[]> prefixScan(final Bytes prefix) {
+        final byte[] prefixEnd = Arrays.copyOf(prefix.get(), prefix.get().length + 1);
+        prefixEnd[prefixEnd.length-1] = (byte)0xFF;
```

Comment:
Sorry, feeling a bit dense... Is this right? It seems like it would cut off some results that should be at the end of the range. 

In particular, it seems like it cuts off keys like `[prefix]FF01`, since that key is strictly greater than `[prefix]FF`.

You might want to take a look at the function that RocksDB uses to compute the range-end for range queries.

Reply:
Yep. I'm taking a look at this now and I don't know what I was thinking. I have tested some of the subMap logic and it indeed seems that I am cutting off some values. Thank you for catching this, ugh.

Range is a bit different because it specifies the entire key, whereas with prefix we're only concerned with the first N bytes of the key. I will think on this some more. I am wondering if it's best to just wrap the entire thing in an iterator that compares the prefix on each "next()" call, and only returns those that meet the full prefix, much like I already have established in the RocksDBDualCFPrefixIterator.

Reply:
Okay, so I believe that I have figured out what I was doing wrong... I needed to do:
`map.subMap(fromKey, true, toKey, false);`
and set toKey to be fromKey + 1 as a byte array. I am currently testing it and it seems to be working, but I want to test further in case I am misunderstanding the submap results.

Reply:
Yeah, that last thing sounds like what I had in mind. The RocksDB function I had in mind is one that basically computes `+1` for an array of bytes (i.e., iterate backwards from the end to find the first byte that isn't already `0xFF` and add `1` to it).